src/administration/integrations/gluescripts.md

```diff
@@ -0,0 +1,248 @@
+
+# GlueScript APIs
+
+## Project API
+
+The project API gives you everything you need to know about the project in the context of which the Gluescript is running. 
+
+For example: `project.environments.master` will give you access to the metadata of the master environment of the current project.
+
+The API starts at `api.project` and is based on a one resource == one object model.
+
+### Properties
+
+The properties of the resource can be access directly on the object, like:
+
+```javascript
+project.owner
+```
+
+### Sub-objects
+
+Sub-objects can be accessed directly on the object, like:
+
+```javascript
+project.settings
+```
+
+### Collections
+
+Resources that are collections support iteration:
+
+```javascript
+project.environments.forEach(function(env) {
+  console.log(env.name);
+});
+```
+
+They also support direct access to an element:
+
+```javascript
+project.environments["master"].name
+```
+
+### Actions
+
+Resource actions are accessible as methods on the object:
+
+```javascript
+project.environments["master"].branch({
+  title="Test",
+  name="test",
+})
+```
+
+## Activity API
+
+The activity API accessed through `activity` gives you all the metadata about the current activity that invoked the Gluescript. The schema is the same as the one for Webhooks, documented here: https://docs.platform.sh/administration/integrations/webhooks.html
+
+## Process API
+The process API gives you access to the environment in the context of which the Gluescript is running .. such as to Environment Variables.
+
+For example:
+
+`process.env.FOO` will give you access to the `FOO` environment variable, this is a great place to access API keys... or other secrets.
+
+## Storage API
+
+A storage API is offered for integrations to store information between calls. E.g. if you want to be able to edit a previous slack message. The storage is scoped per integration.
+
+### `storage.get(key)`: Get a key from the storage
+
+ * `key` (string): The key to get
+
+Returns `null` if the key doesn't exist.
+
+### `storage.set(key, value)`: Create or update a key into the storage
+
+ * `key` (string): The key to set
+ * `value` (string): The value to set, must be a string (will not be cast)
+
+### `storage.remove(key)`: Remove a key from the storage
+
+ * `key` (string): The key to remove
+
+No-op if the key doesn't exist
+
+### `storage.clear()`: Clear the storage
+
+## Fetch API
+
+### fetch(): launch an HTTP request
+
+The fetch API allows you to do HTTP calls to external services. This implements a limited, synchronous version of the Fetch API with a hardcoded timeout of 30s, for example:
+
+```javascript
+var resp = fetch(
+  "http://example.com/",
+  {
+    method: "POST",
+    body: JSON.stringify({
+        "test": "toto",
+    }),
+  }
+)
+
+if (!resp.ok) {
+    throw "unable to fetch";
+}
+```
+See https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch for general information about fetch.
```

Comment:
I don't know why I don't prefer raw string links in the docs, but consider

For more information about fetch, see [this article](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch).

Reply:
When it is something like Mozilla, I do prefer showing the full URL otherwise do something like:

 For more information see this [Mozilla developer network article on Fetch](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch).

or something of the sort